test/widgets/test_widget_init_only.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+import libqtile.widget as widgets
+from libqtile.widget.crashme import _CrashMe
+from libqtile.widget.graph import _Graph
+
+# This file runs a very simple test to check that widgets can be initialised
+# and that keyword arguments are added to default values.
+#
+# This test is not meant to replace any widget specific tests but should catch
+# any mistakes that inadvertently breakag widgets.
+#
+# By default, the test runs on every widget that is listed in __init__.py
+# This is done by building a list called `parameters` which contains a tuple of
+# (widget class, kwargs).
+#
+# Adjustments to the tests can be made below.
+
+# Some widgets require certain parameters to be set when initialising.
+# Widgets listed here will replace the default values
+overrides = [
+    (
+        widgets.BitcoinTicker,
+        {"currency": "USD"}  # Set the currency to prevent locale lookup
```

Comment:
Is there a reason to want to prevent the locale lookup?

Reply:
It made the test fail on my computer. Admittedly, that could just a problem with my locale set up rather than anything else!

Reply:
would be interesting to see the stack trace. I can imagine if it's just LANG_C or something it'll fail, but so will the widget itself

Reply:
```
____________________________________________________________________________________ test_widget_init[BitcoinTicker-kwargs4] _____________________________________________________________________________________

Widget = <class 'libqtile.widget.bitcoin_ticker.BitcoinTicker'>, kwargs = {'dummy_parameter': 1}

    @pytest.mark.parametrize("Widget,kwargs", parameters)
    def test_widget_init(Widget, kwargs):
        # Simple test to check whether widget initialises and sets user-defined args
>       widget = Widget(**kwargs)

test/widgets/test_widget_init_only.py:72:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
libqtile/widget/bitcoin_ticker.py:61: in __init__
    locale.setlocale(locale.LC_MONETARY, "en_US.UTF-8")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

category = 4, locale = 'en_US.UTF-8'

    def setlocale(category, locale=None):

        """ Set the locale for the given category.  The locale can be
            a string, an iterable of two strings (language code and encoding),
            or None.

            Iterables are converted to strings using the locale aliasing
            engine.  Locale strings are passed directly to the C lib.

            category may be given as one of the LC_* values.

        """
        if locale and not isinstance(locale, _builtin_str):
            # convert to string
            locale = normalize(_build_localename(locale))
>       return _setlocale(category, locale)
E       locale.Error: unsupported locale setting

/usr/lib/python3.9/locale.py:610: Error
```

Reply:
I guess you don't have en_US.UTF-8 installed and don't have a default locale?

Reply:
My locale is:
```
LANG=en_GB.UTF-8
LC_CTYPE="en_GB.UTF-8"
LC_NUMERIC="en_GB.UTF-8"
LC_TIME="en_GB.UTF-8"
LC_COLLATE="en_GB.UTF-8"
LC_MONETARY="en_GB.UTF-8"
LC_MESSAGES="en_GB.UTF-8"
LC_PAPER="en_GB.UTF-8"
LC_NAME="en_GB.UTF-8"
LC_ADDRESS="en_GB.UTF-8"
LC_TELEPHONE="en_GB.UTF-8"
LC_MEASUREMENT="en_GB.UTF-8"
LC_IDENTIFICATION="en_GB.UTF-8"
LC_ALL=
```

Reply:
My concern with this is, depending on the setup of the CI machine (and any users running tests at home), it could result in failed tests when the widget itself is fine.

Reply:
I suspect the test framework clears locale settings before actually executing the tests, so we probably need to figure out how to propagate that setting into the tests. IMO it's fine to run with whatever locale users have configured.

Reply:
Sorry - don't think I was clear enough. I inserted the currency line as the test was broken on my laptop and so I assumed it would be broken on CI. I've installed "en_US.UTF-8" locally so the test works on my machine without the override and it appears to work on CI too.